src/osrm/osrm.cpp

```diff
@@ -0,0 +1,22 @@
+#include "osrm/osrm.hpp"
+#include "engine/engine.hpp"
+#include "engine/engine_config.hpp"
+#include "engine/plugins/plugin_base.hpp"
+#include "storage/shared_barriers.hpp"
+#include "util/make_unique.hpp"
+
+namespace osrm
+{
+
+// proxy code for compilation firewall
+OSRM::OSRM(engine::EngineConfig &config_) : engine_(util::make_unique<engine::Engine>(config_)) {}
+
+// needed because unique_ptr needs the size of OSRM_impl for delete
+OSRM::~OSRM() {}
```

Comment:
We could do it properly and make use of non-copyable pimpl or Meyer's pimpl idiom http://oliora.github.io/2015/12/29/pimpl-and-rule-of-zero.html